### PR TITLE
(12/12) [BUGFIX] Remove from protectedDataStore in onDisconnect path

### PR DESCRIPTION
**Pull request by julianknutsen**
New commits start: b10a603

Fixes a bug in the P2PDataStore::onDisconnect path where a PersistablePayload would not be removed from the protectedDataStore.

### CODE REVIEW

**Review by chimp1984 — COMMENTED** [review]
utACK

**Review by freimair — APPROVED** [review]
utAck

**Review by ripcurlx — APPROVED** [review]
utACK - I don't merge this PR as the commits will be part of #3609 anyways. So I'll close it as soon as #3609 gets merged.

**Pull request merged**
